lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1538 @@
+}
+
+function Get-ComputeDriver {
+    if ($buildFor -eq "openstack/compute-hyperv") {
+        $compute_driver = "hyperv.nova.driver.HyperVDriver"
+    }
+    else {
+        $compute_driver = "hyperv.driver.HyperVDriver"
+    }
+    return $compute_driver
+}
+
+function Install-NetworkType {
+    $buildFor = Get-JujuCharmConfig -Scope 'zuul-project'
+    $networkType = Get-JujuCharmConfig -Scope 'network-type'
+    if (($networkType -eq 'ovs') -and ($buildFor -eq 'openstack/networking-hyperv')) {
+        Throw "'$networkType' cannot be used when building for '$buildFor'"
+    }
+    if ($networkType -eq 'hyperv') {
+#        Install-NetworkingHyperV
+        Install-Projects "openstack/networking-hyperv"
+    }
+    elseif ($networkType -eq 'ovs') {
+        Check-OVSPrerequisites
+        Enable-OVS
+        Ensure-InternalOVSInterfaces
+    }
+    else {
+        Throw "Wrong network type config: '$networkType'"
+    }
+}
+
+#function Install-NetworkingHyperV {
+#    Write-JujuLog "Installing networking-hyperv"
+#
+#    $openstackBuild = Join-Path $BUILD_DIR "openstack"
+#    Start-ExecuteWithRetry {
+#        Install-OpenStackProjectFromRepo "$openstackBuild\networking-hyperv"
+#    }
+#    $neutronBin = (Get-CharmServices)['neutron']['binary']
+#    if (!(Test-Path $neutronBin)) {
+#        Throw "$neutronBin was not found."
+#    }
+#}
+
+
+function Install-OVS {
+    param(
+        [Parameter(Mandatory=$true)]
+        [string]$InstallerPath
+    )
+
+    Write-JujuInfo "Running OVS install"
+    $ovs = Get-ManagementObject -Class Win32_Product | Where-Object {$_.Name -match "open vswitch"}
+    if ($ovs){
+        Write-JujuInfo "OVS is already installed"
+        return $true
+    }
+
+    $hasInstaller = Test-Path $InstallerPath
+    if($hasInstaller -eq $false){
+        $InstallerPath = Get-OVSInstaller
+    }
+
+    Write-JujuInfo "Installing from $InstallerPath"
+    $ret = Start-Process -FilePath msiexec.exe -ArgumentList "INSTALLDIR=`"$OVS_DIR`"","/qb","/l*v","$env:APPDATA\ovs-log.txt","/i","$InstallerPath" -Wait -PassThru
+    if($ret.ExitCode) {
+        Throw "Failed to install OVS: $LASTEXITCODE"
+    }
+    Remove-Item $InstallerPath
+    return $true
+}
+
+
+function Import-OVSCertificate {
+	param(
+        [string]$CertificatePath
+    )
+    if (!(Test-Path $CertificatePath)) {
+        $CertificatePath = Get-OVSCertificate
+    }
+    Write-JujuInfo "Importing certificate from $CertificatePath"
+    Import-Certificate $CertificatePath -StoreLocation LocalMachine -StoreName TrustedPublisher
+    Import-Certificate $CertificatePath -StoreLocation LocalMachine -StoreName Root
+    Remove-Item $CertificatePath
+    return $true
+}
+
+
+function Get-OVSInstaller {
+    $urlChecksum = Get-URLChecksum "ovs-installer-url"
+    $location = Get-PackagePath $urlChecksum['URL'] $urlChecksum['CHECKSUM'] `
+                                $urlChecksum['HASHING_ALGORITHM']
+    return $location
+}
+
+function Get-OVSCertificate {
```

Comment:
This is needed for OVS-CI . The generated OVS msi has a self-signed certificate that get's downloaded here for importing.